docs/content/integrations/tableau.mdx

```diff
@@ -151,3 +151,62 @@ defs = Definitions.merge(
     marketing_team_workspace.build_defs(),
 )
 ```
+
+### Add a Data Quality Warning in Tableau using a sensor
+
+When an upstream dependency of a Tableau asset fails to materialize or to pass the asset checks, it is possible to add a [Data Quality Warning](https://help.tableau.com/current/online/en-us/dm_dqw.htm) to the corresponding data source in Tableau. This can be achieved by leveraging the `add_data_quality_warning_to_data_source` in a sensor.
+
+```python file=/integrations/tableau/add-tableau-data-quality-warning.py
+from dagster_tableau import TableauCloudWorkspace
+
+from dagster import (
+    Definitions,
+    EnvVar,
+    RunFailureSensorContext,
+    asset,
+    run_failure_sensor,
+)
+
+# Connect to Tableau Cloud using the connected app credentials
+workspace = TableauCloudWorkspace(
+    connected_app_client_id=EnvVar("TABLEAU_CONNECTED_APP_CLIENT_ID"),
+    connected_app_secret_id=EnvVar("TABLEAU_CONNECTED_APP_SECRET_ID"),
+    connected_app_secret_value=EnvVar("TABLEAU_CONNECTED_APP_SECRET_VALUE"),
+    username=EnvVar("TABLEAU_USERNAME"),
+    site_name=EnvVar("TABLEAU_SITE_NAME"),
+    pod_name=EnvVar("TABLEAU_POD_NAME"),
+)
+
+
+@asset(
+    metadata={"dagster/tableau_data_source_id": "f5660c7-2b05-4ff0-90ce-3199226956c6"}
+)
+def upstream_asset(): ...
+
+
+@run_failure_sensor
+def tableau_run_failure_sensor(
+    context: RunFailureSensorContext, tableau: TableauCloudWorkspace
+):
+    for asset_key in context.dagster_run.asset_selection:
+        data_source_id = upstream_asset.metadata_by_key.get(asset_key, {}).get(
+            "dagster/tableau_data_source_id"
+        )
+        if data_source_id:
+            with tableau.get_client() as client:
+                client.add_data_quality_warning_to_data_source(
+                    data_source_id=data_source_id, message=context.failure_event.message
+                )
+
+
+# We use Definitions.merge to combine the definitions from the Tableau workspace
+# and the Dagster definitions into a single set of definitions to load
+defs = Definitions.merge(
+    workspace.build_defs(),
```

Comment:
it might be a little more clear if we built this definition separately

```
tableau_defs = workspace.build_defs()
upstream_defs = Definitions(assets...)

defs = Definitions.merge(tableau_defs, upstream_defs)
```

Reply:
Done in [e3213d2](https://github.com/dagster-io/dagster/pull/25078/commits/e3213d2d205af5578260a9900ffff9c26010ac0c)